components/raftstore/src/store/peer_storage.rs

```diff
@@ -1238,10 +1199,10 @@ where
         &mut self,
         ctx: &mut InvokeContext,
         snap: &Snapshot,
-        kv_wb: &mut EK::WriteBatch,
-        raft_wb: &mut ER::LogBatch,
-        destroy_regions: &[metapb::Region],
-    ) -> Result<()> {
+        task: &mut WriteTask<EK, ER>,
+        msgs: Vec<RaftMessage>,
```

Comment:
It feels strange to pass `msgs` and `destroy_regions` only for building a result. How about changing it to
```
pub fn apply_snapshot(&mut self, ctx: &mut InvokeContext, snap: &Snapshot, task: &mut WriteTask<EK, ER>, destroy_regions: &[metapb::Region]) -> Result<metapb::Region>;
```